### DISCUSSION TITLE
Understanding and predicting fees

### DISCUSSION THREAD

**Discussion by danielcharrua** [ACINQ/phoenixd#104] [Q&A] [answered]
Hey! I'm linking LNbits with phoenixd as a funding source.

This LBnits instance would be opened to other users, so I'm trying to predict and understand the fees and how this can affect the users inside the instance.

- Is there a way to know/predict what is the total fees to be collected before the users get their sats? As I understand the first payments are used to pay for the liquidity fees, so no payments are shown on my side (if the payment is lower than the fees)

- Is there a way for me to pay upfront the auto-liquidity fees for the 2M channel? I want to accomplish that the users always get their sats, so I'm willing to pay always for the fees to provide service.

- After opening the first channel of 2M sats, can I disable the opening on new channels (auto-liquidity)?

Thank you.

**Comment by tsmith123** [comment]
Asked a similar question here but still waiting for a reply: https://github.com/ACINQ/phoenixd/discussions/88
It would be very useful to know the cutoff point before sats start going to the user.

**Reply by danielcharrua** [reply]
Saw your question earlier without any answers, that's weird 🤔
Maybe @pm47 @dpad85 can help us here 🙏

**Comment by pm47** [comment]
> Is there a way to know/predict what is the total fees to be collected before the users get their sats? As I understand the first payments are used to pay for the liquidity fees, so no payments are shown on my side (if the payment is lower than the fees)

No, it depends on mining fees, which are unpredictable. At best we could expose the cost at a given instant.

> Is there a way for me to pay upfront the auto-liquidity fees for the 2M channel? I want to accomplish that the users always get their sats, so I'm willing to pay always for the fees to provide service.
> After opening the first channel of 2M sats, can I disable the opening on new channels (auto-liquidity)?

Just send funds to your phoenixd until a channel is open, then restart phoenixd with `--auto-liquidity off`

**Reply by danielcharrua** [reply]
@pm47 

> No, it depends on mining fees, which are unpredictable. At best we could expose the cost at a given instant.

Would be nice to expose the fees cost, maybe before opening the channel or add a command into the console to get it at any time.
Example: `/getfees --2M` (for a 2M channel)

> Just send funds to your phoenixd until a channel is open, then restart phoenixd with --auto-liquidity off

So the channel is not opened until the fees are covered? How do I check that? with `/listchannels` or `/getinfo`?

One last scenario, suppose we have a 2M channel opened and incoming is exhausted, if `auto-liquidity` is on, another channel will be opened on a new incoming payment. But in our case using LNbits, the user receiving the new payment will not get any funds, is that right?

Thank you!

**Reply by danielcharrua** [reply]
@pm47 is there a way to turn off the `auto-liquidity` feature from `phoenix.conf`?

**Reply by pm47** [reply]
All options `./phoenixd --name value` can be set in `phoenix.conf` as `name=value`.

**Comment by pm47** [comment, marked as answer]
The main issue here is that phoenixd currently only supports on-the-fly liquidity, whereas you are looking for something more like a one-time manual liquidity request (both are available on the mobile version of Phoenix). Manual liquidity request will soon be available on phoenixd.

In the meantime:

> Would be nice to expose the fees cost, maybe before opening the channel or add a command into the console to get it at any time.

See #107.

> So the channel is not opened until the fees are covered? How do I check that? with /listchannels or /getinfo?

Any of those.

> One last scenario, suppose we have a 2M channel opened and incoming is exhausted, if auto-liquidity is on, another channel will be opened on a new incoming payment.

The existing channel would be extended.

> But in our case using LNbits, the user receiving the new payment will not get any funds, is that right?

At the phoenixd level, any received funds that do not fit in an existing channels, but are not enough to pay for additional liquidity will be added to the fee credit. As to how LNbits manages that, it's a question for LNbits integrators.
